wdl-format/src/token.rs

```diff
@@ -64,27 +89,32 @@ impl<T: Token> IntoIterator for TokenStream<T> {
 }
 
 /// The kind of comment.
-#[derive(Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub enum Comment {
     /// A comment on its own line.
-    Preceding(String),
+    Preceding(Rc<String>),
     /// A comment on the same line as the code preceding it.
-    Inline(String),
+    Inline(Rc<String>),
 }
 
 /// Trivia.
-#[derive(Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub enum Trivia {
     /// A blank line. This may be ignored by the postprocessor.
     BlankLine,
     /// A comment.
     Comment(Comment),
 }
 
-/// Whether optional blank lines are allowed in the current context.
+/// The policy for line spacing.
+///
+/// Blank lines before comments and between comments are always permitted.
 #[derive(Eq, PartialEq, Default, Debug, Clone, Copy)]
 pub enum LineSpacingPolicy {
```

Comment:
Maybe I just haven't hit where this is used, but the comments suggest this applies only to comments. Should the naming reflect that?

Reply:
Yes it should probably have a name change.

This has no effect on `PreToken::BlankLine`s which are always rendered (up to a maximum of one in a row; `[PreToken::BlankLine, PreToken::BlankLine]` would get rendered as a single blank in the `PostToken` stream). `LineSpacingPolicy` is only checked when rendering `Trivia::BlankLine`s.

```suggestion
/// The policy for [`Trivia::BlankLine`] line spacing.
///
/// Blank lines before comments and between comments are always permitted.
#[derive(Eq, PartialEq, Default, Debug, Clone, Copy)]
pub enum TriviaBlankLineSpacingPolicy {
```
maybe?